pkg/api/store/apiservice/filter.go

```diff
@@ -0,0 +1,59 @@
+package apiservice
+
+import (
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/rancher/norman/types"
+	"github.com/rancher/rancher/pkg/clustermanager"
+	apiregistrationv1 "github.com/rancher/types/apis/apiregistration.k8s.io/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+)
+
+type noWatchByAPIServiceStore struct {
+	types.Store
+	manager *clustermanager.Manager
+	version string
+}
+
+func NewAPIServicFilterStoreFunc(cm *clustermanager.Manager, apiVersion string) func(types.Store) types.Store {
+	return func(store types.Store) types.Store {
+		return &noWatchByAPIServiceStore{
+			Store:   store,
+			manager: cm,
+			version: apiVersion,
+		}
+	}
+}
+
+func (s *noWatchByAPIServiceStore) Watch(apiContext *types.APIContext, schema *types.Schema, opt *types.QueryOptions) (chan map[string]interface{}, error) {
+	clustername := s.manager.ClusterName(apiContext)
+	parts := strings.Split(s.version, "/")
```

Comment:
We can extract the version conversion logic(L31-39) to a separate function and add some unit tests for it.

Reply:
Maybe we can use the apiservice name directly and don't need the conversion

Reply:
Those are the api service items in my clusters.
```
v1.                                    Local                        True        25d
v1.apps                                Local                        True        25d
v1.authentication.k8s.io               Local                        True        25d
v1.authorization.k8s.io                Local                        True        25d
v1.autoscaling                         Local                        True        25d
v1.batch                               Local                        True        25d
v1.crd.projectcalico.org               Local                        True        1h
v1.monitoring.coreos.com               Local                        True        1h
v1.networking.k8s.io                   Local                        True        25d
v1.rbac.authorization.k8s.io           Local                        True        25d
v1.storage.k8s.io                      Local                        True        25d
v1beta1.admissionregistration.k8s.io   Local                        True        25d
v1beta1.apiextensions.k8s.io           Local                        True        25d
v1beta1.apps                           Local                        True        25d
v1beta1.authentication.k8s.io          Local                        True        25d
v1beta1.authorization.k8s.io           Local                        True        25d
v1beta1.batch                          Local                        True        25d
v1beta1.certificates.k8s.io            Local                        True        25d
v1beta1.coordination.k8s.io            Local                        True        25d
v1beta1.events.k8s.io                  Local                        True        25d
v1beta1.extensions                     Local                        True        25d
v1beta1.metrics.k8s.io                 kube-system/metrics-server   True        25d
v1beta1.policy                         Local                        True        25d
v1beta1.rbac.authorization.k8s.io      Local                        True        25d
v1beta1.scheduling.k8s.io              Local                        True        25d
v1beta1.storage.k8s.io                 Local                        True        25d
v1beta2.apps                           Local                        True        25d
v2beta1.autoscaling                    Local                        True        25d
v2beta2.autoscaling                    Local                        True        25d
v3.management.cattle.io                Local                        True        6d
v3.project.cattle.io                   Local                        True        1h
```

Most of them would be fine with api service name. But the one `v1.` is not validated. So I still need to use split